ambry-messageformat/src/main/java/com.github.ambry.messageformat/BlobStoreHardDelete.java

```diff
@@ -80,24 +80,36 @@ public MessageInfo getMessageInfo(Read read, long offset, StoreKeyFactory storeK
       if (headerFormat.hasEncryptionKeyRecord()) {
         deserializeBlobEncryptionKey(stream);
       }
+      short lifeVersion = 0;
+      if (headerFormat.hasLifeVersion()) {
+        lifeVersion = headerFormat.getLifeVersion();
+      }
```

Comment:
If we don't adopt MessageHeader_Format_V3, do we still need to check this?  I am considering if lifeVersion is not put in header, we probably need to remove `hasLifeVersion` method from V1/V2 as well.

Reply:
Sorry, ignore this, I feel like we do need to lifeVersion in message header